### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by diebland**

### CODE REVIEW

**Comment by AnaAGG** [comment]
# lab-customer-analysis-round-6 and final round

Hi Blandine 🙋🏻‍♀️! Let's go with some coments about you labs!

- Firstly, good job using functions! It is a good practice 🔝
- Only as a detail, when you create the correlation matix remember that you can also use the `heatmap` from `seaborn` library. This method give us the same information that `corr` method but more friendly. 
    ```python
    ax = sns.heatmap(df, annot=True)
    ```
- Regardless the outlier question: *it seems that it didn't work out. But I don't know why.* Your code was almost perfect, it was just a detail. One possibility to solve this problem is: 
    ```python
    #create a list with the name of the columns where we detect outliers
    outliers = ['customer_lifetime_value', 'monthly_premium_auto', 'total_claim_amount', 'number_of_policies']
    def clean_outliers(data, outliers):
        for column in outliers:
            iqr = np.percentile(data[column],75) - np.percentile(data[column],25)
            upper_limit = np.percentile(data[column],75) + 1.5*iqr
            lower_limit = np.percentile(data[column],25) - 1.5*iqr
            data.loc[data[column] > upper_limit, column] = upper_limit
            data.loc[data[column] < lower_limit, column] = lower_limit
        return data

    df2 = clean_outliers(df, outliers) # this return the dataset cleaned without outliers. 

    # ⚠️ The only thing that I did is remove the if condition and pass only the name of target columns. 
    ```
- The error that you get when fit the model. This is because you are passing categorical variables without conversion using HotEncoding or getdummies. Sklearn models dont understand the string varibles, for this is mandatory preform the conversion tools. So... what do you should do?
  - Apply the get dummies to all the categorical variables
    ```python
        # First we create all the predictors variables in a subdataframe
        dummy = df.drop('total_claim_amount', axis=1)

        # By default the get_dummies only works in categorical variables even if we have also numerical. 
        cat_df = pd.get_dummies(dummy, drop_first=True)

        # When we have all the variables ready, the next step is define the X, Y and perform the train test split 

        X = cat_df
        y = df['total_claim_amount']

        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=10)

        # Finally, fit the model
        model = LinearRegression()
        model.fit(X_train, y_train) 
    ```
So... until now what we learn? In this first part of the bootcamp we learn to perform a basic machine learning model. As a summary:
- Load the data
- Explore the data in our dataset
    - Check missing values
    - Correlation matrix among variables
    - Check numeric and categorical variables
    - Check the outliers and remove them
    - Normalize (numerical variables) and One Hot/Label Encoding(categorical)
- Train-test split
- Perform different model and check what is the best, for this we can use:
  - R2
  - mse
  - rmse
  - mae


Really good job throughout this part of the bootcamp Blandine. You have done really clearly work, with your jupyters very organised and clean. In addition, you have been internalising all the concepts that we have been learning during these first few weeks. Congratulations!!! 🚀

Nevertheless, if you have any question, dont hesitase to ask me 😉

image
